pyerrors/input/sfcf.py

```diff
@@ -29,266 +53,210 @@ def read_sfcf(path, prefix, name, **kwargs):
-    names -- Alternative labeling for replicas/ensembles. Has to have the appropriate length
-    ens_name : str
-        replaces the name of the ensemble
-    """
-
-    if kwargs.get('im'):
-        im = 1
-        part = 'imaginary'
-    else:
-        im = 0
-        part = 'real'
-
-    if kwargs.get('b2b'):
-        b2b = 1
-    else:
-        b2b = 0
-
-    T = 0
-    ls = []
     for (dirpath, dirnames, filenames) in os.walk(path):
         ls.extend(dirnames)
         break
     if not ls:
         raise Exception('Error, directory not found')
     # Exclude folders with different names
-    for exc in ls:
-        if not fnmatch.fnmatch(exc, prefix + '*'):
-            ls = list(set(ls) - set([exc]))
+    if "replica" in kwargs:
+        ls = reps
+    else:
+        for exc in ls:
+            if not fnmatch.fnmatch(exc, prefix + '*'):
+                ls = list(set(ls) - set([exc]))
     if len(ls) > 1:
         ls.sort(key=lambda x: int(re.findall(r'\d+', x[len(prefix):])[0]))  # New version, to cope with ids, etc.
     replica = len(ls)
+    print('Read', part, 'part of', name, 'from', prefix[:-1], ',', replica, 'replica')
+    idl = []
     if 'names' in kwargs:
         new_names = kwargs.get('names')
+        if len(new_names)!=len(set(new_names)):
+            raise Exception("names are nor unique!")
         if len(new_names) != replica:
             raise Exception('Names does not have the required length', replica)
     else:
         # Adjust replica names to new bookmarking system
         new_names = []
         for entry in ls:
-            idx = entry.index('r')
+            try:
+                idx = entry.index('r')
+            except:
+                raise Exception("Automatic recognition of replicum failed, please enter the key word 'names'.")
+        
             if 'ens_name' in kwargs:
                 new_names.append(kwargs.get('ens_name') + '|' + entry[idx:])
             else:
                 new_names.append(entry[:idx] + '|' + entry[idx:])
-
-    print('Read', part, 'part of', name, 'from', prefix[:-1], ',', replica, 'replica')
     for i, item in enumerate(ls):
         sub_ls = []
-        for (dirpath, dirnames, filenames) in os.walk(path + '/' + item):
-            sub_ls.extend(filenames)
-            break
-        for exc in sub_ls:
-            if not fnmatch.fnmatch(exc, prefix + '*'):
-                sub_ls = list(set(sub_ls) - set([exc]))
-        sub_ls.sort(key=lambda x: int(re.findall(r'\d+', x)[-1]))
-
-        first_cfg = int(re.findall(r'\d+', sub_ls[0])[-1])
-
-        last_cfg = len(sub_ls) + first_cfg - 1
-
-        for cfg in range(1, len(sub_ls)):
-            if int(re.findall(r'\d+', sub_ls[cfg])[-1]) != first_cfg + cfg:
-                last_cfg = cfg + first_cfg - 1
+        if "files" in kwargs:
+            sub_ls = kwargs.get("files")
+            sub_ls.sort(key=lambda x: int(re.findall(r'\d+', x)[-1]))
```

Comment:
Does this work in combination with the automatic name recognition? You could filter first, before you try to detect the name from the content of ls, to throw out all of the stuff, that you don't like anyways.

Reply:
If I understand this correctly, the answer is: The name recognition is dependent on the replicum names, "files" just filters the content in each replicum, so yes, this should work as far as I am aware (but I tested if `files` without `names` nonetheless).
In any case, a feature that is not implemented yet is something like files that can be different for each replicum, this was rather low on my priority list for now...